src/Features/Core/Portable/Completion/CompletionRules.cs

```diff
@@ -65,9 +82,32 @@ public virtual bool MatchesFilterText(CompletionItem item, string filterText, Co
                 return false;
             }
 
-            var patternMatcher = this.GetPatternMatcher(_completionService.GetCultureSpecificQuirks(filterText));
+            return GetMatchWithWorkaround(item, filterText) != null;
+        }
+
+        // Hack for Turkish:
```

Comment:
Agreed. I'm not thrilled with the method name either. This is how we're defining the behavior. We're not really hacking or working around and issue. It's just that our logic has a fallback for the turkish locale.